tools/readme-catalog/gen-cloudevent-catalog.js

```diff
@@ -0,0 +1,66 @@
+import * as fs from 'fs';
+import { dirname } from 'path';
+import { fileURLToPath } from 'url';
+import { getGeneratedStringWithReplacement } from './common.js';
+
+/**
+ * Generates a CloudEvent catalog for the README.
+ */
+
+/**
+ * Gets markdown for the proto / JSON schema table cell.
+ * @param {object} schemaEntry The JSON schema catalog entry.
+ * @returns {string} Markdown for the schema entry.
+ */
+const getSchemaLinkMarkdown = (schemaEntry) => {
+  // Get the path the to the data from the datatype.
+  // in: "google.events.cloud.audit.v1.LogEntryData"
+  // out: "cloud/audit/v1"
+  const prefix = 'google.';
+  const schemaPath = schemaEntry.datatype.substring(
+    schemaEntry.datatype.indexOf(prefix) + prefix.length,
+    schemaEntry.datatype.indexOf(`.${schemaEntry.name}`),
+  ).replace(/\./g, '/');
```

Comment:
nit: A direct replace statement might be more clear.

```suggestion
  ).replace('.', '/');
```

Reply:
This replacement would result in an error.

`String.prototype.replace` only catches the first match.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

In this case, we want to replace all `.` with `/` characters in the schema path. I've usually seen this pattern (change the substring to a simple regex) works.